forage-android/src/main/java/com/joinforage/forage/android/core/element/state/ElementState.kt

```diff
@@ -1,20 +1,39 @@
 package com.joinforage.forage.android.core.element.state
 
 import com.joinforage.forage.android.core.element.ElementValidationError
+import com.joinforage.forage.android.model.USState
 
-data class ElementState(
+data class ElementState<InputDetails>(
```

Comment:
@devinmorgan  Will this be a breaking change for the client? My guess is no, but just want to confirm.

i.e. will the client have to update their code to conform to the new interfaces?

```kotlin
// client code
// before:

ForageElement

// after (breaking change):
ForageElement<ClientInsertsDetailsTypeHere>
```

Also, should we explicitly declare `public` classes as `public` or is it fine to omit? Not sure what the kotlin linting rules prefer.

Reply:
I fear for the flakiness of the QA tests so I am taking a screenshot to capture that the CI tests passed for this PR at one point (see below).

> Will this be a breaking change for the client? 
So as you can see the CI tests against this PR is that I did not change the Sample App in this PR. The significance of them passing is that this PR is reverse compatible (for the with the Sample App at least). The reason this change is reverse compatible with the Sample App is because the Sample App never explicitly refers to any of the types that this PR changes (e.g. `ElementState`, `AbstractElement`, etc). If the Sample App had referenced these types then this would have been a breaking change.

This same point applies to GoPuff. If they currently reference the types that this PR does change instead of letting the compiler infer the types, then this PR would be a breaking change. However, even if GoPuff does reference the types changed, it would only require minor updates in their code (just updating the types)

![image](https://github.com/teamforage/forage-android-sdk/assets/12377418/393955d9-0f8f-44ff-8003-6f1d4043946d)